### PR TITLE
[DB-477] [risk=no] fetch concept name of id/ code matches

### PR DESCRIPTION
**Pull request by SrushtiGangireddy**
1. In the case of concept id/ code match fetch concept name for the medline plus link to return meaningful search.

### CODE REVIEW

File: public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java [hunk 1]
```diff
@@ -561,12 +561,14 @@ public SurveyQuestionResult apply(AchillesResult o) {
             String conceptId = String.valueOf(con.getConceptId());
 
             if((con.getStandardConcept() == null || !con.getStandardConcept().equals("S") ) && (searchConceptsRequest.getQuery().equals(conceptCode) || searchConceptsRequest.getQuery().equals(conceptId))){
-                response.setMatchType(conceptCode.equals(searchConceptsRequest.getQuery()) ? MatchType.CODE : MatchType.ID );
-
                 List<Concept> stdConcepts = conceptDao.findStandardConcepts(con.getConceptId());
                 response.setStandardConcepts(stdConcepts.stream().map(TO_CLIENT_CONCEPT).collect(Collectors.toList()));
             }
 
+            if(!Strings.isNullOrEmpty(searchConceptsRequest.getQuery()) && (searchConceptsRequest.getQuery().equals(conceptCode) || searchConceptsRequest.getQuery().equals(conceptId))) {
```
Comment (calbach) [inline public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java:568]: Can multiple concepts match this if statement? Or they should be unique by code and concept ID?

File: public-ui/src/app/views/ehr-view/ehr-view.component.ts [hunk 2]
```diff
@@ -304,4 +308,11 @@ export class EhrViewComponent implements OnInit, OnDestroy {
   public changeResults(e) {
     this.loadPage();
   }
+
+  public getTerm() {
+    if (this.matchType === 'CONCEPT_ID' || this.matchType === 'CONCEPT_CODE') {
```
Comment (calbach) [inline public-ui/src/app/views/ehr-view/ehr-view.component.ts]: nit: type `MatchType`, compare against `MatchType.CONCEPT_ID`

File: public-ui/src/app/views/ehr-view/ehr-view.component.ts [hunk 3]
```diff
@@ -175,6 +177,8 @@ export class EhrViewComponent implements OnInit, OnDestroy {
     this.searchResult.items = this.searchResult.items.filter(
       x => this.dbc.TO_SUPPRESS_PMS.indexOf(x.conceptId) === -1);
     this.items = this.searchResult.items;
+    this.matchType = this.searchResult.matchType.toString();
```
Comment (calbach) [inline public-ui/src/app/views/ehr-view/ehr-view.component.ts]: nit: do you need these variables? Seems like you could just directly reference `this.searchResult.matchType` below

**Review by calbach — APPROVED** [review]

File: public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java (hunk 1 above)
Comment (SrushtiGangireddy) [reply public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java:568]: They should be unique by code and concept id. I guess no concept would match both code and concept id.

**Pull request merged**